app/src/renderer/connectors/lcdClientMock.js

```diff
@@ -891,6 +916,142 @@ module.exports = {
+        results.push(txResult(1, `Amount of ${coin.denom}s cannot be negative`))
+        return results
+      } else if (!coinBalance || coinBalance.amount < depositCoinAmt) {
+        results.push(txResult(1, `Not enough ${coin.denom}s in your account`))
+        return results
+      }
+
+      coinBalance.amount -= depositCoinAmt
+
+      // ============= TOTAL PROPOSAL's DEPOSIT =============
+      // Increment total deposit of the proposal
+      let depositIndex = proposal.total_deposit.findIndex(
+        deposit => deposit.denom === coin.denom
+      )
+      if (depositIndex === -1) {
+        // if there's no previous deposit in that denom we just append it to the total deposit
+        proposal.total_deposit.push(coin)
+      } else {
+        // if there's an existing deposit with that denom we add the submited deposit amount to it
+        let newAmt = String(
+          parseInt(proposal.total_deposit[depositIndex].amount) +
+            parseInt(coin.amount)
+        )
+        proposal.total_deposit[depositIndex].amount = newAmt
+      }
+
+      // ============= USER'S DEPOSITS =============
+      // check if there's an existing deposit by the depositer
+      let prevDepositIndex = state.deposits[proposal_id].findIndex(
+        deposit => deposit.depositer === depositer
+      )
+
+      if (prevDepositIndex === -1) {
+        // if no previous deposit by the depositer, we add it to the existing deposits
+        state.deposits[proposal_id].push(submittedDeposit)
+        break // break since no need to iterate over other coins
+      } else {
+        // ============= USER'S DEPOSITS WITH SAME COIN DENOM =============
+        // if there's a prev deposit, add the new amount to the corresponding coin
+        let prevDepCoinIdx = state.deposits[proposal_id][
+          prevDepositIndex
+        ].amount.findIndex(prevDepCoin => {
+          return prevDepCoin.denom === coin.denom
+        })
+        if (prevDepCoinIdx === -1) {
+          state.deposits[proposal_id][prevDepositIndex].amount.push(coin)
+        } else {
+          // there's a previous deposit from the depositer with the same coin
+          let newAmt =
+            parseInt(
+              state.deposits[proposal_id][prevDepositIndex].amount[
+                prevDepCoinIdx
+              ].amount
+            ) + parseInt(coin.amount)
+
+          state.deposits[proposal_id][prevDepositIndex].amount[
+            prevDepCoinIdx
+          ].amount = String(newAmt)
+        }
+      }
+    }
+    // TODO: double check if we need to update it inside the loop
+    incrementSequence(fromAccount)
+
+    // check if the propoposal is now active
+    if (proposal.proposal_status === `Pending`) {
+      // TODO: get min deposit coin from gov params instead of stake params
+      let depositCoinAmt = proposal.total_deposit.find(coin => {
+        return coin.denom === `steak`
+      }).amount
+      if (parseInt(depositCoinAmt) >= 10) {
```

Comment:
>= 10 is the min deposit?

Reply:
yup, but maybe because is an `sdk.Dec` I need to multiply it ? thoughts ?

Reply:
yeah but it's an `sdk.Dec`, do I need to multiply the amount as well ?

Reply:
I was only implying, that this could use a comment or taking the value of the min deposit from a global constant.

Reply:
> taking the value of the min deposit from a global constant.

Indeed, it's actually part of the governance params, which aren't yet merged on the SDK. I'll add a comment

Reply:
#1564